src/posts/introduction-webc.md

```diff
@@ -0,0 +1,236 @@
+
+To illustrate how this can be useful, let’s create a card component that needs to have a title, some media, and a brief description. We won’t concern ourselves with the CSS for this component, but let’s assume we’re doing some fancy things that require us to place the aformentione title, media, and description inside some wrappers. We can define `a-card.webc` like this:
+
+```html
+<article class="card">
+  <div class="card__container">
+    <div class="card__media">
+      <slot name="media"></slot>
+    </div>
+    <strong class="card__title">
+      <slot name="title"></slot>
+    </strong>
+    <div class="card__body">
+      <slot name="body"></slot>
+    </div>
+  </div>
+</article>
+```
+
+When we use this card in our markup we place the title, media, and description as children of our custom element and give them the appropriate `slot` attribute.
+
+```html
+<a-card>
+  <a href="#" slot="title">The Hypnotoad</a>
+  <img src="hypnotoad.gif" slot="media">
+  <p slot="body">All glory to the Hypnotoad~</p>
+</a-card>
+```
+
+Notice that we can order our elements however we want. We don’t have to place the media above the title when we define the card, because they will get pulled into the appropriate order when they're slotted into the component.
+
+## WebC for web components
+
+Up to this point, we’ve been looking at how we can use WebC in Eleventy for the same kinds of tasks for which we’ve previously used one of Eleventy’s template languages. But the WebC README declares:
+
+> WebC is for Single File Web Components
+
+So what does all of this have to do with web components? Well, we can use WebC single file components as a convenient way to write and include web components — with templates and shadow DOM — in our sites. As a simple example, suppose we want to implement a theme toggle on our site that allows visitors to choose a light or dark theme, or to use whichever matches their operating system. We’re implementing this as a web component because it relies on JavaScript to switch the theme. Without JavaScript, the buttons wouldn’t do anything, so we don’t want them to be visible unless JavaScript is available.
+
+We’ll call our component `theme-select.webc`:
+
+```html
+<script>
+  class ThemeSelect extends HTMLElement {
+    connectedCallback() {
+      const template = this.querySelector("template");
+
+      // Swap out the template with its contents so they become visible
+      template.replaceWith(template.content);
+    }
+  }
+
+  customElements.define("theme-select", ThemeSelect);
+</script>
+
+<template>
+  <button>Light</button>
+  <button>System</button>
+  <button>Dark</button>
+</template>
+
+<style>
+  theme-select {
+    /* ... */
+  }
+</style>
+```
+
```

Comment:
Include a blockquote callout here to note that the class format and `connectedCallback()` are native web component constructs and link to MDN docs that best fit perhaps?